docs/recipes/automate-release-workflow.md

```diff
@@ -1,79 +1,94 @@
-var log = require('gulplog');
-var git = require('gulp-git');
-var fs = require('fs');
-
-gulp.task('changelog', function () {
-  return gulp.src('CHANGELOG.md', {
-    buffer: false
-  })
-    .pipe(conventionalChangelog({
-      preset: 'angular' // Or to any other commit message convention you use.
-    }))
-    .pipe(gulp.dest('./'));
-});
-
-gulp.task('github-release', function(done) {
-  conventionalGithubReleaser({
-    type: "oauth",
-    token: 'abcdefghijklmnopqrstuvwxyz1234567890' // change this to your own GitHub token or use an environment variable
-  }, {
-    preset: 'angular' // Or to any other commit message convention you use.
-  }, done);
-});
-
-gulp.task('bump-version', function () {
-// We hardcode the version change type to 'patch' but it may be a good idea to
-// use minimist (https://www.npmjs.com/package/minimist) to determine with a
-// command argument whether you are doing a 'major', 'minor' or a 'patch' change.
-  return gulp.src(['./bower.json', './package.json'])
-    .pipe(bump({type: "patch"}).on('error', log.error))
-    .pipe(gulp.dest('./'));
-});
-
-gulp.task('commit-changes', function () {
-  return gulp.src('.')
-    .pipe(git.add())
-    .pipe(git.commit('[Prerelease] Bumped version number'));
-});
-
-gulp.task('push-changes', function (done) {
-  git.push('origin', 'master', done);
-});
-
-gulp.task('create-new-tag', function (done) {
-  var version = getPackageJsonVersion();
-  git.tag(version, 'Created Tag for version: ' + version, function (error) {
-    if (error) {
-      return done(error);
-    }
-    git.push('origin', 'master', {args: '--tags'}, done);
+The recipe below bumps the project version, commits the changes to git and creates a new GitHub release.
+
+For publishing a GitHub release you'll need to [create a personal access token](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token) and add it to your project. However, we don't want to commit it, so we'll use [`dotenv-safe`](https://www.npmjs.com/package/dotenv-safe) to load it from a git-ignored `.env` file.
+
+Create an `.env.example` file to define required environment variables:
+
+```
+GH_TOKEN=
+```
+
+We'll commit this, so leave it empty. Next, duplicate this file, rename it to `.env` and add your GitHub token:
+
+```
+GH_TOKEN=ff34885...
+```
+
+Finally, add `.env` to your `.gitignore`.
+
+Moving on, install all the necessary dependencies for this recipe:
+
+```sh
+npm install --save-dev conventional-recommended-bump conventional-changelog-cli conventional-github-releaser dotenv-safe execa pify
+```
+
+Finally, `gulpfile.js`:
+
+``` js
+const gulp = require('gulp');
+const conventionalRecommendedBump = require('conventional-recommended-bump');
+const conventionalGithubReleaser = require('conventional-github-releaser');
+const execa = require('execa');
+const fs = require('fs/promises');
+const pify = require('pify');
+const dotenv = require('dotenv-safe');
```

Comment:
Probably just use `dotenv`?  I looked into this package and the downloads are small and not much benefit over just `dotenv`.

Reply:
Sure, I personally like how with `dotenv-safe` you can define which environment variables the build process expects should exist, but here it's fine either way.